### PR TITLE
Make VSTest repo buildable in VMR non-source-build

### PR DESCRIPTION
**Pull request by ViktorHofer**
Contributes to https://github.com/dotnet/source-build/issues/4030

- DotNetBuildFromSource -> DotNetBuildSourceOnly
- DotNetBuildFromSourceFlavor -> DotNetBuildOrchestrator
- SourceBuild.props -> DotNetBuild.props
- AfterSolutionBuild.targets: Add a condition as runtime download currently doesn't work inside the VMR

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -8,14 +8,14 @@
     necessary as the output of this leg is used in other CI source build legs. Those could be
     targeting NetCurrent or NetPrevious hence we must produce both.
   -->
-  <PropertyGroup Condition=" '$(DotNetBuildFromSource)' == 'true' AND '$(DotNetBuildFromSourceFlavor)' != 'Product' ">
+  <PropertyGroup Condition=" '$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuild)' != 'true' ">
```
Comment (mmitche) [inline Directory.Build.targets:11]:
```suggestion
  <PropertyGroup Condition=" '$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuildOrchestrator)' != 'true' ">
```

This pattern is about repo level source build vs. building as the whole product. Should be the above.

File: Directory.Build.targets [hunk 2]
```diff
@@ -8,14 +8,14 @@
...
     <TargetFrameworks>$(NetPrevious);$(NetCurrent)</TargetFrameworks>
   </PropertyGroup>
 
   <!--
     Source build the product: this is the all up build of the product which needs only NetCurrent
   -->
-  <PropertyGroup Condition=" '$(DotNetBuildFromSource)' == 'true' AND '$(DotNetBuildFromSourceFlavor)' == 'Product' ">
+  <PropertyGroup Condition=" '$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuild)' == 'true' ">
```
Comment (mmitche) [inline Directory.Build.targets:18]:
```suggestion
  <PropertyGroup Condition=" '$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuildOrchestrator)' == 'true' ">
```

File: src/Microsoft.TestPlatform.Build/Microsoft.TestPlatform.Build.csproj [hunk 3]
```diff
@@ -11,9 +11,9 @@
   <PropertyGroup>
     <!-- This package is included even in source build -->
     <IsPackable>true</IsPackable>
-    <NuspecFile Condition=" '$(DotNetBuildFromSource)' != 'true' ">Microsoft.TestPlatform.Build.nuspec</NuspecFile>
-    <NuspecFile Condition=" '$(DotNetBuildFromSource)' == 'true' AND '$(DotNetBuildFromSourceFlavor)' != 'Product' ">Microsoft.TestPlatform.Build.sourcebuild.nuspec</NuspecFile>
-    <NuspecFile Condition=" '$(DotNetBuildFromSource)' == 'true' AND '$(DotNetBuildFromSourceFlavor)' == 'Product' ">Microsoft.TestPlatform.Build.sourcebuild.product.nuspec</NuspecFile>
+    <NuspecFile Condition="'$(DotNetBuildSourceOnly)' != 'true'">Microsoft.TestPlatform.Build.nuspec</NuspecFile>
+    <NuspecFile Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuild)' != 'true'">Microsoft.TestPlatform.Build.sourcebuild.nuspec</NuspecFile>
```
Comment (mmitche) [inline src/Microsoft.TestPlatform.Build/Microsoft.TestPlatform.Build.csproj:15]:
```suggestion
    <NuspecFile Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuildOrchestrator)' != 'true'">Microsoft.TestPlatform.Build.sourcebuild.nuspec</NuspecFile>
```

File: src/Microsoft.TestPlatform.Build/Microsoft.TestPlatform.Build.csproj [hunk 4]
```diff
@@ -11,9 +11,9 @@
...
+    <NuspecFile Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuild)' == 'true'">Microsoft.TestPlatform.Build.sourcebuild.product.nuspec</NuspecFile>
```
Comment (mmitche) [inline src/Microsoft.TestPlatform.Build/Microsoft.TestPlatform.Build.csproj:16]:
```suggestion
    <NuspecFile Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuildOrchestrator)' == 'true'">Microsoft.TestPlatform.Build.sourcebuild.product.nuspec</NuspecFile>
```

File: src/package/Microsoft.TestPlatform.CLI/Microsoft.TestPlatform.CLI.csproj [hunk 5]
```diff
@@ -4,9 +4,9 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <NuspecFile Condition=" '$(DotNetBuildFromSource)' != 'true' ">Microsoft.TestPlatform.CLI.nuspec</NuspecFile>
-    <NuspecFile Condition=" '$(DotNetBuildFromSource)' == 'true' AND '$(DotNetBuildFromSourceFlavor)' != 'Product' ">Microsoft.TestPlatform.CLI.sourcebuild.nuspec</NuspecFile>
-    <NuspecFile Condition=" '$(DotNetBuildFromSource)' == 'true' AND '$(DotNetBuildFromSourceFlavor)' == 'Product' ">Microsoft.TestPlatform.CLI.sourcebuild.product.nuspec</NuspecFile>
+    <NuspecFile Condition="'$(DotNetBuildSourceOnly)' != 'true' ">Microsoft.TestPlatform.CLI.nuspec</NuspecFile>
+    <NuspecFile Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuild)' != 'true'">Microsoft.TestPlatform.CLI.sourcebuild.nuspec</NuspecFile>
+    <NuspecFile Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuild)' == 'true'">Microsoft.TestPlatform.CLI.sourcebuild.product.nuspec</NuspecFile>
```
Comment (mmitche) [inline src/package/Microsoft.TestPlatform.CLI/Microsoft.TestPlatform.CLI.csproj:9]:
```suggestion
    <NuspecFile Condition="'$(DotNetBuildSourceOnly)' == 'true' and '$(DotNetBuildOrchestrator)' == 'true'">Microsoft.TestPlatform.CLI.sourcebuild.product.nuspec</NuspecFile>
```

**Review by mmitche — APPROVED** [review]
Changes otherwise look good. Should at least be straight compatible in source-build mode.

File: Directory.Build.targets (hunk 1 above)
Comment (ViktorHofer) [reply Directory.Build.targets:11]: Repo level source build doesn't set DotNetBuild though, or does it?

**Review by nohwnd — APPROVED** [review]

**Pull request merged**